core/lib/queued_job_processor/src/lib.rs

```diff
@@ -127,6 +147,10 @@ pub trait JobProcessor: Sync + Send {
             job_id,
             error_message
         );
+
+        if attempts == Some(self.max_attempts()) {
+            METRICS.max_attempts_reached[&(Self::SERVICE_NAME, format!("{job_id:?}"))].inc();
```

Comment:
Stupid question: Can't having `job_id` as a label lead to a huge number of time series for different metrics? Or do we assume that running out of attempts is rare enough? I *think* having job ID logged with a `WARN` / `ERROR` level and/or represented as [a metric exemplar](https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#exemplars) would be more appropriate. (I'm not that familiar with exemplars though, and `vise` currently doesn't support them, although they [are supported](https://docs.rs/prometheus-client/latest/prometheus_client/metrics/exemplar/index.html) by the underlying library.)

Reply:
This should be rare enough. For production envs I think this is completely ok, but for staging it could be a problem in theory. 
The reason why I want to keep it is that having job id as a label allows us to display it in the alert message which is pretty useful. I'd go with label and change it later if it ever becomes an issue